addons/mail/static/src/core/web/recipients_input.js

```diff
@@ -183,7 +183,7 @@ export class RecipientsInput extends Component {
             tags.push({
                 id: recipient.partner_id,
                 canEdit: true,
-                text: recipient.name || recipient.email,
+                text: recipient.name || recipient.email || _t("Unknown"),
                 name: recipient.name,
```

Comment:
Would prefer to make it so that we have
```suggestion
                name: recipient.name || _t("Unnamed"),
```

Instead of the changes you did in the popover. This way you don't touch too much the logic there and only adapt what is given to it.

You can also apply this comment to the `many2many_tags_email` widget.